EvolvSwiftSDK/EvolvSwiftSDKTests/Common-Tests/EvolvClientTests.swift

```diff
@@ -435,3 +435,120 @@ extension EvolvClientTests {
         XCTAssertTrue(expectedSubmittedData.isSubset(of: actualSubmittedData))
     }
 }
+
+// MARK: - On listeners
+extension EvolvClientTests {
+    func testOnContextValueAdded() {
+        let options = EvolvClientOptions(evolvDomain: "participants-stg.evolv.ai", participantID: "80658403_1629111253538", environmentId: "4a64e0b2ab", analytics: true, remoteContext: [:], beacon: evolvBeacon)
+        let client = EvolvClientImpl(options: options, evolvAPI: evolvAPI, scope: scope).initialize().wait()
+        
+        struct Value: Equatable, Hashable {
+            let key: String!
+            let value: String!
+        }
+        
+        var actualValuesAdded = Set<Value>()
+        client.on(topic: CONTEXT_VALUE_ADDED) { value in
+            actualValuesAdded.insert(Value(key: value["key"] as? String, value: value["value"] as? String))
+        }
+        _ = client.set(key: "location", value: "US", local: false)
+        _ = client.set(key: "view", value: "home", local: true)
+        
+        let expectedValuesAdded: Set<Value> = [.init(key: "location", value: "US"), .init(key: "view", value: "home")]
+        
+        XCTAssert(expectedValuesAdded.isSubset(of: actualValuesAdded))
+    }
+    
+    func testOnContextValueChanged() {
+        let options = EvolvClientOptions(evolvDomain: "participants-stg.evolv.ai", participantID: "80658403_1629111253538", environmentId: "4a64e0b2ab", analytics: true, remoteContext: [:], beacon: evolvBeacon)
+        let client = EvolvClientImpl(options: options, evolvAPI: evolvAPI, scope: scope).initialize().wait()
+        
+        struct Value: Equatable, Hashable {
+            let key: String!
+            let value: String!
+        }
+        
+        var actualValuesAdded = Set<Value>()
+        client.on(topic: CONTEXT_VALUE_ADDED) { value in
+            actualValuesAdded.insert(Value(key: value["key"] as? String, value: value["value"] as? String))
+        }
+        client.on(topic: CONTEXT_VALUE_CHANGED) { value in
+            actualValuesAdded.insert(Value(key: value["key"] as? String, value: value["value"] as? String))
```

Comment:
This would incorreclty pass if the update fired and added event instead - can you use a seperate actualValues list for Changed?

Reply:
addressed it here PR #27